File: CS/DxSample/Editors/CheckedListBoxExControl.cs
using DevExpress.XtraEditors;
using System.ComponentModel;

namespace DxSample.Editors {
    public class CheckedListBoxExControl :CheckedListBoxControl {
        private bool fUseWholeItemToChangeCheckedStateByMouse = true;
        [DefaultValue(true)]
        public bool UseWholeItemToChangeCheckedStateByMouse {
            get { return this.fUseWholeItemToChangeCheckedStateByMouse; }
            set { this.fUseWholeItemToChangeCheckedStateByMouse = value; }
        }

        protected override ListBoxControlHandler CreateHandler() {
            return new CheckedListBoxExControlHandler(this);
        }
    }
}

File: VB/DxSample/Editors/CheckedListBoxExControl.vb
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace DxSample.Editors
    Public Class CheckedListBoxExControl
        Inherits CheckedListBoxControl

        Private fUseWholeItemToChangeCheckedStateByMouse As Boolean = True
        <DefaultValue(True)> _
        Public Property UseWholeItemToChangeCheckedStateByMouse() As Boolean
            Get
                Return Me.fUseWholeItemToChangeCheckedStateByMouse
            End Get
            Set(ByVal value As Boolean)
                Me.fUseWholeItemToChangeCheckedStateByMouse = value
            End Set
        End Property

        Protected Overrides Function CreateHandler() As ListBoxControlHandler
            Return New CheckedListBoxExControlHandler(Me)
        End Function
    End Class
End Namespace
